strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/test/java/org/carlspring/strongbox/nuget/NuspecTest.java

```diff
@@ -0,0 +1,491 @@
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpression;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+import org.carlspring.strongbox.nuget.NugetFormatException;
+import org.carlspring.strongbox.nuget.Nuspec;
+import org.carlspring.strongbox.nuget.Reference;
+import org.carlspring.strongbox.nuget.metadata.Dependency;
+import org.carlspring.strongbox.nuget.metadata.Framework;
+import org.carlspring.strongbox.nuget.metadata.VersionRange;
+import org.junit.jupiter.api.Test;
+import org.w3c.dom.Document;
+import org.xml.sax.SAXException;
+
+import org.semver.Version;
+
+/**
+ * @author Dmitry Sviridov
+ */
+public class NuspecTest
+{
+
+    /**
+     * Extracts data from an XML document
+     *
+     * @param xmlData
+     *            XML document
+     * @param xPath
+     *            xPath expression
+     * @return string representation of data
+     * @throws XPathExpressionException
+     *             incorrect xPath expression
+     * @throws ParserConfigurationException
+     *             XML parser creation error
+     * @throws SAXException
+     *             invalid XML
+     * @throws IOException
+     *             data reading error
+     */
+    private String getStringValue(byte[] xmlData,
+                                  String xPath)
+        throws XPathExpressionException,
+        SAXException,
+        IOException,
+        ParserConfigurationException
+    {
+        DocumentBuilderFactory domFactory = DocumentBuilderFactory.newInstance();
+        domFactory.setNamespaceAware(true);
+        DocumentBuilder builder = domFactory.newDocumentBuilder();
+        Document doc = builder.parse(new ByteArrayInputStream(xmlData));
+        NamespaceContext context = new CustomContext();
+        XPathFactory pathFactory = XPathFactory.newInstance();
+        XPath path = pathFactory.newXPath();
+        path.setNamespaceContext(context);
+        XPathExpression expr = path.compile(xPath);
+        return (String) expr.evaluate(doc, XPathConstants.STRING);
+    }
+
+    /**
+     * Test parsing file specifications from XML
+     *
+     * @throws Exception
+     *             error during the test
+     */
+    @Test
+    public void testParseMethod()
+        throws Exception
+    {
+        final String fileName = "nuspec/test.nuspec.xml";
+        Nuspec result = Nuspec.parse(NugetTestResourceUtil.getAsStream(fileName));
+        assertEquals("Neolant.ProjectWise.IsolationLevel.Implementation", result.getId(), "Package ID");
+        assertEquals(Version.parse("1.4.7.550"), result.getVersion(), "Package Version");
```

Comment:
I think we will have to translate these messages to English. Non-russian speakers will have a hard time reading these exceptions. (Applicable across the entire pull request).

Reply:
I translated most of the texts. This one I left intentionally as it was not meaningful for the test response and it's also part of the nuspec resource. Note that the assertion message is translated though.

I'll try to translate these, ok.